### ISSUE TITLE
Recipe Book

### ISSUE THREAD

**Issue by Hanse00** [Hanse00/CrafTech-old#33]
I've been thinking a little about ways for us to stand out from other tech mods, and I think it could be an interesting idea to have a sort of recipe book, basically it'd work the following way:
- All machines have to be crafted in a special machine crafting table
- The table can hold a recipe book, and can only craft the machines for which there are recipes in the book for
- The book is an inventory that can hold recipe pages (Similar to mystcraft notebook) :book: 
- Pages are obtained by villager trade, and dungeon loot (for the time being) :page_facing_up: 

This way we add some sort of barrier into the crafting of machines, rather than them being craft able just because you have the resources.
Later pages could be obtained through a research system, building onto the technological aspect.

I'd like any possible feedback on this idea.

**Comment by tterrag1098** [comment]
Sounds good, as long as the research isn't "You must spend time for no reason other than spending time" like some of the other research systems feel like. In general though, sounds like a very good idea.

**Comment by Hanse00** [comment]
I've gotten around to implement the special crafting table for this, but I'm a little uncertain about how we want to set up the inventory for this.

The way I see it, there's two options:
- Like a vanilla crafting table (can only contain things as long as the interface is open)
- Similar to the RP2 table (stores inventory in the tile at all times)

What do you think we should do @yonsaber?

**Comment by tterrag1098** [comment]
The only reason vanilla crafting tables don't have an inventory is to cut down on ticking blocks in the world. As long as you don't expect this table to be spammed as much as a regular crafting table, I see no reason for it not to have an inventory. Personally I would lean towards that option, more like the RP2 or Tcon tables.

**Comment by yonsaber** [comment]
I agree with @tterrag1098. I see no reason for it not to use the inventory, having the ability to 'store' items in the table as you are crafting will be a huge bonus. Also if we want to have large multiblock machines then creating lots of the same block will be needed so having a storage in a table will help out.

**Comment by Hanse00** [comment]
Alright, I'll look into setting this up, I imagine the TConstruct source code might he helpful here (:heart: open source).

Also I think @tterrag1098 deserve a great thanks for the helpful comments provided to this project, it really means a lot to us that someone is taking their time to chime in.

**Comment by tterrag1098** [comment]
Heh, np. I'd help out coding too if I didn't already waste too much time on my own mods ;)

**Comment by Hanse00** [comment]
You're welcome to fork any day :smile:

But really, just the comments are an amazingly useful resource. It helps to be able to break out of your own bubble of thought, and be able to see the things from a different perspective too.
